src/components/dataTable/js/tableCellDirective.js

```diff
@@ -0,0 +1,213 @@
+}
+
+MdTableCellController.prototype.initCtrl = function(tableCtrl) {
+  this.table = tableCtrl;
+
+  this.$scope.$watch(this.getAlignment.bind(this), function(newAlignment, oldAlignment) {
+
+    if (this.validAlignments.indexOf(newAlignment) !== -1) {
+      this.$element.addClass('md-align-' + newAlignment);
+    }
+
+    if (this.validAlignments.indexOf(oldAlignment) !== -1 && newAlignment !== oldAlignment) {
+      this.$element.removeClass('md-align-' + oldAlignment);
+    }
+
+  }.bind(this));
+};
+
+MdTableCellController.prototype.getAlignment = function() {
+
+  // Transform NodeList into an Array.
+  var headRows = [].slice.call(this.table.getHead().rows);
+  var cellIndex = this.$element[0].cellIndex;
+
+  return headRows
+    .map(function(row) {
+      return angular.element(row.cells[cellIndex]).data('align');
+    })
+    .filter(function(cell) {
+      return !!cell;
+    })[0];
+};
+
+function MdTableAlignDirective() {
+
+  return  {
+    restrict: 'A',
+    require: ['mdCell', '^^mdHead'],
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs) {
+    attrs.$observe('mdAlign', function(align) {
+      element.data('align', align);
+    });
+  }
+}
+
+function MdTableOrderByDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdOrderBy', 'mdCell', '^^mdOrder', '^^mdTable'],
+    scope: {},
+    controller: MdTableOrderByController,
+    controllerAs: '$mdOrderBy',
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var cellCtrl = ctrls[1];
+    var orderCtrl = ctrls[2];
+    var tableCtrl = ctrls[3];
+
+    ctrl.initCtrl(tableCtrl, orderCtrl, cellCtrl);
+  }
+}
+
+function MdTableOrderByController($element, $scope, $attrs, $compile, $mdUtil, $$mdSvgRegistry) {
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$attrs = $attrs;
+  this.$compile = $compile;
+  this.$mdUtil = $mdUtil;
+  this.$$mdArrowUpIcon = $$mdSvgRegistry.mdArrowUp;
+
+  this.isActiveRegex = new RegExp('^-?' + this.$attrs.mdOrderBy + '$');
+
+  this.$attrs.$observe('mdOrderBy', function (orderBy) {
+    if (orderBy) {
+      this.enableSorting();
+    } else {
+      this.disableSorting();
+    }
+  }.bind(this));
+}
+
+MdTableOrderByController.prototype.initCtrl = function(tableCtrl, orderCtrl, cellCtrl) {
+  this.table = tableCtrl;
+  this.orderCtrl = orderCtrl;
+  this.cellCtrl = cellCtrl;
+
+  this.unregisterWatch = null;
+  this.clickListenerFn = this.updateSortOrder.bind(this);
+
+  // Watch for position changes to realign the sort icon.
+  this.$scope.$watch(function() {
+    return this.cellCtrl.$element[0].cellIndex;
+  }.bind(this), function() {
+    var icon = this.$element[0].querySelector('.md-sort-icon');
+
+    if (icon) {
+      if (this.$element.data('align') === 'right') {
+        this.$element.prepend(icon);
+      } else {
+        this.$element.append(icon);
+      }
+    }
+
+  }.bind(this));
```

Comment:
ES6 would be definitely nice. But as long as we support IE11, we are not able to use it without a transpiler.

Also ES6 classes would be great ;) but it would not solve the context issue (still needing a bind, self or arrow fn)